### PR TITLE
feat: status_code attribute for Response, HTTPError, HTTPStatus

### PR DESCRIPTION
**Pull request by kgriffs**
This patch adds a status_code attribute (property) to the Response, HTTPError,
and HTTPStatus classes to afford introspection by middleware and error
handlers.

### CODE REVIEW

File: falcon/http_error.py [hunk 1]
```diff
@@ -46,7 +46,10 @@ class HTTPError(Exception):
         error in a future version of falcon.
 
     Args:
-        status (str): HTTP status code and text, such as "400 Bad Request"
+        status (Union[str,int]): HTTP status code or line (e.g.,
```
Comment (CaselIT) [inline falcon/http_error.py:49]: we don't seem to handle the case of int/httpstatus here, like in `self.title = title or status`

File: falcon/http_status.py [hunk 2]
```diff
@@ -25,14 +26,18 @@ class HTTPStatus(Exception):
...
+            ``'400 Bad Request'``). This may be set to a member of
+            :class:`http.HTTPStatus`, an HTTP status line string or byte
+            string (e.g., ``'200 OK'``), or an ``int``.
         headers (dict): Extra headers to add to the response.
         text (str): String representing response content. Falcon will encode
             this value as UTF-8 in the response.
 
     Attributes:
-        status (str): HTTP status line, e.g. '748 Confounded by Ponies'.
+        status (Union[str, int]): The HTTP status line or integer code for
```
Comment (CaselIT) [inline falcon/http_status.py:38]:
```suggestion
        status (Union[str,int]): The HTTP status line or integer code for
```

**Review by CaselIT — COMMENTED** [review]
considering typing, I think it would be nicer if `status` had a narrower type, but there has always been the possibility of setting multiple types so nothing new here.

for the rest it seems ok

File: falcon/http_error.py (hunk 1 above)
Comment (kgriffs) [reply falcon/http_error.py:49]: Added tests and a fix for this.

**Review by CaselIT — APPROVED** [review]

**Comment by kgriffs** [comment]
> considering typing, I think it would be nicer if `status` had a narrower type, but there has always been the possibility of setting multiple types so nothing new here.
> 
> for the rest it seems ok

Ideally we would have added `status_code` before this and left `status` alone, but I'd like to just keep the consistency for now and avoid breaking changes; we can consider changing things up in the next major release.

**Review by vytas7 — APPROVED** [review]
LGTM :+1:

**Pull request merged**